### PR TITLE
docs: add diagrams for concept introduction

### PR DESCRIPTION
**Pull request by eeberhard**
## Description

I had this minor improvement on a branch for a while, decided to rebase and push since we have several new people reading the docs now.

Note: I also changed the deployment workflow run triggers.

## Supporting information

SVGs with dark and light mode support

---

Screenshot 2023-12-09 at 15 26 16
---
Screenshot 2023-12-09 at 15 26 24
---

## Review guidelines

Estimated Time of Review: 3 minutes

## Checklist before merging:

- [x] Confirm that the relevant changelog(s) are up-to-date in case of any user-facing changes

### CODE REVIEW

**Review by JasperTan97 — APPROVED** [review]
Adorable drawings!

File: .github/workflows/build-docs.yml [hunk 1]
```diff
@@ -1,15 +1,11 @@
...
-      - ".github"
-      - ".github/workflows/*-docs.yml"
   push:
+    branches:
+      - main
     paths:
-      - ".github"
-      - ".github/workflows/*-docs.yml"
-      - "docs/static/schemas/**"
+      - "docs/**"
```
Comment (domire8) [inline .github/workflows/build-docs.yml:8]: What happened in the workflows?

File: docs/src/css/custom.css [hunk 2]
```diff
@@ -33,3 +33,14 @@
 [data-theme='dark'] .footer {
   background-color: var(--ifm-background-accent-color);
 }
+
+/*[data-theme='light'] .themedSVG [fill='#FFFF50'] {*/
```
Comment (domire8) [inline docs/src/css/custom.css:37]: Do we need the commented lines?

**Review by domire8 — COMMENTED** [review]
Nice!

**Review by alexialechot — APPROVED** [review]
Well done! 👏

File: .github/workflows/build-docs.yml (hunk 1 above)
Comment (eeberhard) [reply .github/workflows/build-docs.yml:8]: well it was rebuilding on push / PR on the .github paths. I think it was leftover from prototyping but that doesn't make much sense really

File: docs/src/css/custom.css (hunk 2 above)
Comment (eeberhard) [reply docs/src/css/custom.css:37]: no, my mistake

**Review by jjenscodee — APPROVED** [review]

**Pull request merged**